src/analysis/utils/runtime-utils.ts

```diff
@@ -74,15 +75,15 @@ export type NormalizedComputeRegion = NominalType<string, 'ComputeRegion'>;
 // TODO: test when zone and region have types
 export const getNormalizedComputeRegion = (config: GoogleRuntimeConfig | AzureConfig): NormalizedComputeRegion => {
   if (isGceConfig(config) || isGceWithPdConfig(config)) {
-    return getRegionFromZone(config.zone).toUpperCase() as NormalizedComputeRegion;
+    return (getRegionFromZone(config.zone) || defaultComputeRegion).toUpperCase() as NormalizedComputeRegion;
   }
   if (isDataprocConfig(config)) {
-    return config.region.toUpperCase() as NormalizedComputeRegion;
+    return (config.region || defaultComputeRegion).toUpperCase() as NormalizedComputeRegion;
   }
   if (isAzureConfig(config)) {
-    return config.region.toUpperCase() as NormalizedComputeRegion;
+    return (config.region || defaultAzureComputeRegion).toUpperCase() as NormalizedComputeRegion;
```

Comment:
is this method ever used in an Azure context? All I can find is Environments.ts, and that's specifically for showing runtime region.